recommender/recommender.py

```diff
@@ -999,11 +1023,35 @@ def parse_xml(cls, node, runtime, keys, _id_generator):  # pylint: disable=unuse
 
         """
         block = runtime.construct_xblock_from_class(cls, keys)
-        lines = []
-        for line in node.text.split('\n'):
-            line = line.strip()
-            if len(line) > 2:
-                lines.append(json.loads(line))
 
-        block.default_recommendations = data_structure_upgrade(lines)
+        node_xml = etree.tostring(node, encoding='utf-8')
+        try:
+            dom = md.parseString(node_xml.encode('utf-8'))
+        except Exception:
+            raise UpdateFromXmlError("An error occurred while parsing the XML content.")
```

Comment:
Why would you swallow the actual exception, only to re-raise a more generic exception? If you don't know how to handle an exception, don't catch it at all, so that the parent function has the opportunity to catch it and handle it correctly.